crates/database/src/orders.rs

```diff
@@ -547,7 +557,8 @@ WHERE
         WHEN 'buy' THEN sum_buy < buy_amount
     END AND
     (NOT invalidated) AND
-    (NOT presignature_pending);
+    (NOT presignature_pending) AND
+    (class <> 'limit' OR surplus_fee <> 0);
```

Comment:
This should probably only include limit orders with fairly recent `surplus_fee` quotes in case the `LimitOrderQuoter` is experiencing some issues for a while.

Reply:
https://github.com/cowprotocol/services/pull/714/commits/9c5465f39d6887f90895765db5885e295e7b12d8